ddtrace/contrib/aiohttp/middlewares.py

```diff
@@ -29,24 +31,32 @@ def attach_context(request):
         service = app[CONFIG_KEY]['service']
         distributed_tracing = app[CONFIG_KEY]['distributed_tracing_enabled']
 
+        context = tracer.context_provider.active()
+
+        # Create a new context based on the propagated information.
+        # Do not fill context with distributed sampling if the tracer is disabled
+        # because the would call the callee to generate references to data which
+        # has never been sent to agent.
+        # [TODO:christian] this is quite generic and applies to any similar library so
+        # at some point we should have some shared code which populates context from headers.
+        if tracer.enabled and distributed_tracing:
+            trace_id = int(request.headers.get(PARENT_TRACE_HEADER_ID, 0))
+            parent_span_id = int(request.headers.get(PARENT_SPAN_HEADER_ID, 0))
+            sampling_priority = request.headers.get(SAMPLING_PRIORITY_HEADER_ID)
```

Comment:
Sure? Because what we want is:

- header set to 1 -> sampling_priority == 1
- header set to 0 -> sampling_priority == 0
- header unset -> sampling_priority == None

But with this one liner I think if the header is set to 0 we get request.headers.get(...,0) -> this is 0, so then the or evaluates its right member and returns None (instead of 0). I tried it, it breaks a unitary test. Having sampling_priority to None or 0 is a very different case, when None, it means we should use the previous signature sampler, if set to 0 it means the trace should not be sent according to priority sampling policy.

Thanks however for zooming on that point, it happens there's a little caveat below. Updating.

Reply:
Oh in that case yes, the oneliner doesn't apply. Keep it that way so.